### PR TITLE
Add check to test tool outputs using pytest

### LINKED ISSUE #12817: Add output validation for test tool to FIM module
|Wazuh version|Component|
|---|---|
| 4.4.0+| Wazuh FIM |

## Description
Hello team, this issue is to add the validation of the output from a run of the FIM test tool. For that we can use Pytest in order to use their tools to validate the output. The test tool generates output files in json format, so the script to test these outputs should be able to read a JSON file, after it should compare Test tool output with known values. It implies:

- [x] Refactor utils.py file to have maintainability over time
- [x] Make python files compatible with the PEP8 standard
- [x] Add documentation to every function inside new and old python files
- [x] Add pytest in order to check test tool outputs

## Output example:
Atomic action: `{"action":"StartTransaction","result":true}`

Transaction execution: `{
    "data":
    [
        {
            "Operation type": "INSERTED",
            "action": "SyncTxnRows",
            "value":
            [
                {
                    "attributes": "10",
                    "checksum": "a2fbef8f81af27155dcee5e3927ff6243593b91a",
                    "dev": 2221,
                    "gid": 0,
                    "group_name": "root",
                    "hash_md5": "4b531524aa13c8a54614100b570b3dc7",
                    "hash_sha1": "7902feb66d0bcbe4eb88e1bfacf28befc38bd58b",
                    "hash_sha256": "e403b83dd73a41b286f8db2ee36d6b0ea6e80b49f02c476e0a20b4181a3a062a",
                    "inode": 18277083,
                    "last_event": 1596489275,
                    "mode": 0,
                    "mtime": 1578075431,
                    "options": 131583,
                    "path": "/tmp/test_1.txt",
                    "perm": "-rw-rw-r--",
                    "scanned": 1,
                    "size": 4925,
                    "uid": 0,
                    "user_name": "fakeUser"
                }
            ]
        },
        {
            "Operation type": "INSERTED",
            "action": "SyncTxnRows",
            "value":
            [
                {
                    "attributes": "10",
                    "checksum": "a2fbef8f81af27155dcee5e3927ff6243593b91a",
                    "dev": 2221,
                    "gid": 0,
                    "group_name": "root",
                    "hash_md5": "4b531524aa13c8a54614100b570b3dc7",
                    "hash_sha1": "7902feb66d0bcbe4eb88e1bfacf28befc38bd58b",
                    "hash_sha256": "e403b83dd73a41b286f8db2ee36d6b0ea6e80b49f02c476e0a20b4181a3a062a",
                    "inode": 18277083,
                    "last_event": 1596489275,
                    "mode": 0,
                    "mtime": 1578075431,
                    "options": 131583,
                    "path": "/tmp/test_2.txt",
                    "perm": "-rw-rw-r--",
                    "scanned": 1,
                    "size": 4925,
                    "uid": 0,
                    "user_name": "fakeUser"
                }
            ]
        },
        {
            "Operation type": "DELETED",
            "action": "GetDeletedRows",
            "options":
            {},
            "value":
            {
                "path": "/tmp/test_1.txt"
            }
        }
    ]
}`

### PR DESCRIPTION
**Pull request by FrancoRivero**
|EPIC|Component|
|---|---|
| https://github.com/wazuh/wazuh/issues/9103 | FIM |

|Issue|
|---|
| Closes #12817  | FIM |

## Description
Hello team, there is a project in order to add checks after running the test tool. 
Taking advantage of this change I modified code structure, documented missing functions, formatted the code with google docs and PEP8 standard.
This development has two parts for one side, we have these changes and for other side we will have a change in [wazuh-jenkins](https://github.com/wazuh/wazuh-jenkins) in order to add `pytest `and `pytest-html` with these dependencies we are able to run these tests inside our Jenkins check.
Regards

## Logs/Alerts example
When we run a RTR test after running ASAN check with these results will check JSON files:
image
We will have a report in HTML in order to see clearly the test outputs:
image
image
## Tests
- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Valgrind (memcheck and descriptor leaks check)
  - [x] AddressSanitizer
- Memory tests for Windows
  - [x] Scan-build report

### CODE REVIEW

**Review by antoniomanuelfr — APPROVED** [review]

**Comment by FrancoRivero** [comment]
image
These are the changes named in the last commit

**Review by MarcelKemp — APPROVED** [review]

**Review by chemamartinez — CHANGES_REQUESTED** [review]

**Review by chemamartinez — APPROVED** [review]

**Review by antoniomanuelfr — APPROVED** [review]

**Review by jotacarma90 — APPROVED** [review]
LGTM

**Review by jotacarma90 — APPROVED** [review]
Reviewed and tested!
LGTM

**Review by chemamartinez — APPROVED** [review]

File: src/Readme.md [hunk 1]
```diff
@@ -30,38 +30,44 @@ make TARGET=server|agent DEBUG=1 TEST=1
...
-| `--scheck`              | Run AStyle on the code for checking purposes. Example: `python3 build.py --scheck <data_provider\|shared_modules/dbsync\|shared_modules/rsync\|shared_modules/utils\|wazuh_modules/syscollector>` |
-| `--sformat`             | Run AStyle on the code formatting the needed files. Example: `python3 build.py --sformat <data_provider\|shared_modules/dbsync\|shared_modules/rsync\|shared_modules/utils\|wazuh_modules/syscollector>` |
-| `--scanbuild` SCANBUILD | Run scan-build on the code. Example: `python3 build.py --scanbuild <agent\|server\|winagent>` |
+| `-r`, `--readytoreview` | Run all the quality checks needed to create a PR. Example: `python3 build.py -r <data_provider\|shared_modules/dbsync\|shared_modules/rsync\|shared_modules/utils\|wazuh_modules/syscollector\|syscheckd>` |
+| `-rc`, `--readytoreviewandclean` | Run all the quality checks needed to create a PR and clean results. Example: `python3 build.py -r <data_provider\|shared_modules/dbsync\|shared_modules/rsync\|shared_modules/utils\|wazuh_modules/syscollector\|syscheckd>` |
```
Comment (Dwordcito) [inline src/Readme.md:46]: Use Single letter arguments.

https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

File: src/build.py [hunk 2]
```diff
@@ -1,108 +1,134 @@
...
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-r", "--readytoreview",
+                        help="Run all the quality checks needed to \
+                              create a PR. Example: python3 build.py \
+                              -r <{}>".format(utils.moduleList()))
+    parser.add_argument("-d", "--deleteLogs",
+                        help="Clean log results. Example: python3 \
+                              build.py -d <{}>".format(utils.moduleList()))
+    parser.add_argument("-rc", "--readytoreviewandclean",
```
Comment (Dwordcito) [inline src/build.py:39]: ditto

**Review by Dwordcito — COMMENTED** [review]
Please remove the line breaks.
image

**Review by jnasselle — APPROVED** [review]

**Pull request merged**